src/main.ts

```diff
@@ -622,23 +622,25 @@ export class Purchases {
         window.history.pushState({ checkoutOpen: true }, "");
       }
 
-      const onClose =
-        this._flags.rcSource === "app"
-          ? undefined
-          : () => {
-              const event = createCheckoutSessionEndClosedEvent();
-              this.eventsTracker.trackSDKEvent(event);
-              window.removeEventListener("popstate", onClose as EventListener);
-
-              if (component) {
-                unmount(component);
-              }
-
-              certainHTMLTarget.innerHTML = "";
-
-              Logger.debugLog("Purchase cancelled by user");
-              reject(new PurchasesError(ErrorCode.UserCancelledError));
-            };
+      const shouldPassOnCloseBehaviour =
+        this._flags.rcSource === "app" || this._flags.rcSource === "embedded";
```

Comment:
```suggestion
      const shouldPassOnCloseBehaviour =
        ["app", "embedded"].includes(this._flags.rcSource);
```

And maybe put the array up as a more visible constant?

Reply:
Done